plugins/modules/net_tools/gandi_livedns.py

```diff
@@ -0,0 +1,215 @@
+  values:
+    description:
+    - The record values.
+    - Required for C(state=present).
+    type: list
+    elements: str
+    aliases: [ content ]
+  zone:
+    description:
+    - The name of the Zone to work with (e.g. "example.com").
+    - The Zone must already exist.
+    - Either I(zone) or I(domain) is required.
+    type: str
+  domain:
+    description:
+    - The name of the Domain to work with (e.g. "example.com").
+    - Either I(zone) or I(domain) is required.
+    type: str
+'''
+
+EXAMPLES = r'''
+- name: Create a test A record to point to 127.0.0.1 in the my.com zone.
+  gandi_livedns:
+    zone: my.com
+    record: test
+    type: A
+    values:
+    - 127.0.0.1
+    api_key: dummyapitoken
+  register: record
+
+- name: Create a mail CNAME record to www.my.com domain
+  gandi_livedns:
+    domain: my.com
+    type: CNAME
+    record: mail
+    values:
+    - www
+    api_key: dummyapitoken
+    state: present
+
+- name: Change its TTL
+  gandi_livedns:
+    domain: my.com
+    type: CNAME
+    values:
+    - www
+    ttl: 10800
+    api_key: dummyapitoken
+    state: present
+
+- name: Delete the record
+  gandi_livedns:
+    domain: my.com
+    type: CNAME
+    record: mail
+    api_key: dummyapitoken
+    state: absent
+'''
+
+RETURN = r'''
+record:
+    description: A dictionary containing the record data.
+    returned: success, except on record deletion
+    type: complex
+    contains:
+        values:
+            description: The record content (details depend on record type).
+            returned: success
+            type: list
+            sample:
+            - 192.0.2.91
+            - 192.0.2.92
+        record:
+            description: The record name.
+            returned: success
+            type: str
+            sample: www
+        ttl:
+            description: The time-to-live for the record.
+            returned: success
+            type: int
+            sample: 300
+        type:
+            description: The record type.
+            returned: success
+            type: str
+            sample: A
+        domain:
+            description: The domain associated with the record.
+            returned: success
+            type: str
+            sample: my.com
+        zone:
+            description: The zone associated with the record.
+            returned: success
+            type: str
+            sample: my.com
+'''
+
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible_collections.community.general.plugins.module_utils.gandi_livedns_api import GandiLiveDNSAPI
+
+
+def main():
+    module = AnsibleModule(
+        argument_spec=dict(
+            api_key=dict(type='str', required=True, no_log=True),
+            record=dict(type='str', default='@', aliases=['name']),
+            state=dict(type='str', default='present', choices=['absent', 'present']),
+            ttl=dict(type='int', default=None),
```

Comment:
I don't remember if you can enforce it with ansible but the ttl has to be greater than 300

Reply:
I didn't enforce the minimal value, but now the module returns an explicit error message in case of invalid params:

TASK [community.general.gandi_livedns] ********************************************************
fatal: [localhost]: FAILED! => changed=false 
  msg: 'API Error Bad request: rrset_ttl :100 is less than minimum value 300'